### ISSUE TITLE
Provision command always fails with "field label not supported: spec.externalName"

### ISSUE THREAD

**Issue by mr-flannery** [kubernetes-sigs/service-catalog#2740]
# Bug Report 

**What happened**:

When running any kind of `svcat provision` command, e.g. `svcat feature-flags-instance --plan lite --class feature-flags`, it fails with `Error: unable to search classes by name (field label not supported: spec.externalName)`.

After playing around with `-v` I also got this:

```
svcat -v 999 provision feature-flags-instance --plan lite --class feature-flags
I1029 14:30:42.237907   67111 loader.go:359] Config loaded from file:  /Users/<USER>/.kube/config
I1029 14:30:42.241669   67111 round_trippers.go:419] curl -k -v -XGET  -H "User-Agent: svcat/v0.0.0 (darwin/amd64) kubernetes/$Format" -H "Authorization: Bearer <TOKEN>" -H "Accept: application/json, */*" 'https://<HOST>/apis/servicecatalog.k8s.io/v1beta1/clusterserviceclasses?fieldSelector=spec.externalName%3Dfeature-flags'
I1029 14:30:42.386451   67111 round_trippers.go:438] GET https://<HOST>/apis/servicecatalog.k8s.io/v1beta1/clusterserviceclasses?fieldSelector=spec.externalName%3Dfeature-flags 400 Bad Request in 144 milliseconds
I1029 14:30:42.386509   67111 round_trippers.go:444] Response Headers:
I1029 14:30:42.386529   67111 round_trippers.go:447]     Cache-Control: no-cache, private
I1029 14:30:42.386540   67111 round_trippers.go:447]     Content-Type: application/json
I1029 14:30:42.386551   67111 round_trippers.go:447]     Content-Length: 159
I1029 14:30:42.386561   67111 round_trippers.go:447]     Date: Tue, 29 Oct 2019 13:30:42 GMT
I1029 14:30:42.386743   67111 request.go:947] Response Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"field label not supported: spec.externalName","reason":"BadRequest","code":400}
Error: unable to search classes by name (field label not supported: spec.externalName)
```

**What you expected to happen**:

I expected the service to be provisioned.

**How to reproduce it (as minimally and precisely as possible)**:

This might be difficult. I have a [Gardener](https://gardener.cloud/) cluster running on GCP with which I try to consume services from SAP Cloud Platform. There's an app running there called "service manager" that lists/exposes the service brokers running for the services there. I installed service catalog into my cluster alongside [service-broker-proxy-k8s](https://github.com/Peripli/service-broker-proxy-k8s/issues). As far as I understand, the service broker proxy registers the brokers running on SAP Cloud Platform with the service catalog installation. Anyway, that part works (works as in when running `svcat mp` I see services).

Now, when trying to provision a service, it fails.

**Anything else we need to know?**:

To be honest, I have no idea where the error lies. Maybe the service brokers are somehow registered incorrectly or something. Unfortunately, I cannot provide more information, since the service broker proxy does not log anything when the provision command fails and I don't have access to the logs of service brokers nor of the service manager (the additional thing running on SAP Cloud Platform mentioned above), whereas the `catalog-service-manager` logs a lot of stuff, but nothing I could match to the failing command...

My hope here is that someone with knowledge of the service catalog cli can hazard an educates guess about where the error might be coming from. 

Side-note: some documentation for the `-v` flag would be nice. `-v, --v Level             number for the log level verbosity` is only of limited help, since there's no info about the levels. I just guessed numbers untli something happened. Also most unix commands use `-v` as shorthand for `verbose`, whereas your use of `-v` rather reminds me of loglevels (but that's subjective).

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.0", GitCommit:"2bd9643cee5b3b3a5ecbd3af49d09018f0773c77", GitTreeState:"clean", BuildDate:"2019-09-19T13:57:45Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```
- service-catalog version:
```
Client Version: v0.2.2
Server Version: v1.16.2
```
- Cloud provider or hardware configuration:
Gardener cluster on GCP (if that helps...)
- Do you have api aggregation enabled? 
  - Do you see the configmap in kube-system? 
  - Does it have all the necessary fields?
    - `kubectl get cm -n kube-system extension-apiserver-authentication -o yaml` and look for `requestheader-XXX` fields
```requestheader-extra-headers-prefix: '["X-Remote-Extra-"]'
  requestheader-group-headers: '["X-Remote-Group"]'
  requestheader-username-headers: '["X-Remote-User"]'
```
- Install tools:
  - Did you use helm? What were the helm arguments? Did you `--set` any extra values?
I followed https://kubernetes.io/docs/tasks/service-catalog/install-service-catalog-using-helm/ with helm v2.15
- Are you trying to use ALPHA features? Did you enable them?
No, at least not that I'm aware of.

**Comment by mszostok** [comment]
/reopen
/remove-lifecycle rotten

I will take a look if that problem occurs in 0.3.0 release.

**Comment by jhvhs** [comment]
This bug was fixed in v0.3.0
